docs/data/data-grid/pagination/pagination.md

```diff
@@ -122,7 +150,96 @@ React.useEffect(() => {
 
 {{"demo": "ServerPaginationGrid.js", "bg": "inline"}}
 
-### Cursor implementation
+#### Unknown row count
+
+When the row count is not available, you can use the following steps to enable server-side pagination:
+
+- Set the prop `paginationMode` to `server`
+- Set the `rowCount` as `-1` using one of the above mentioned ways to let the Grid know that the row count is not available
+- Since the `rowCount` is unknown, the Grid needs to know whether more records are available on the server. Pass the `paginationMeta.hasNextPage` boolean property to indicate whether more records are available.
+- Use the `onPaginationModelChange` prop to react to the `paginationModel` changes
```

Comment:
These sections are a bit hard to read, how about we organize everything into a table, and keep shorter descriptions for each of the scenario?
If I understand this correctly, the table should look like this:

|                              | Known row count | Unknown row count | Estimated row count |
| :--------------------------- | :-------------- | :---------------- | :------------------ |
| `rowCount`                   | `number`        | `-1`              | `-1`                |
| `paginationMeta.hasNextPage` | —               | `boolean`         | `boolean`           |
| `estimatedRowCount`          | —               | —                 | `number`            |

<details>
<summary>Raw markdown</summary>

```
|                              | Known row count | Unknown row count | Estimated row count |
| :--------------------------- | :-------------- | :---------------- | :------------------ |
| `rowCount`                   | `number`        | `-1`              | `-1`                |
| `paginationMeta.hasNextPage` | —               | `boolean`         | `boolean`           |
| `estimatedRowCount`          | —               | —                 | `number`            |
```
</details>

What do you think?

Reply:
The `paginationMode` and `onPaginationModelChange` can be covered once in the "Index-based pagination" section too.

Reply:
I agree it's a bit cluttered, thanks for the suggestions. I'll try to accommodate them.

Reply:
@cherniavskii I did some revamping thanks to your suggestions, however, it still doesn't feel super clear readability-wise. 😞 

I am thinking of dividing this page into sub-pages. For now:

- Pagination
  - Overview
  - Server-side pagination
  - Recipes
  
The recipes section could be started with this example: https://mui.com/x/react-data-grid/pagination/#custom-pagination-ui, we could add more later.

Would it be an overkill though? Wdyt?